src/Types.jl

```diff
@@ -157,14 +157,13 @@ PackageSpec(n::AbstractString, u::UUID, v::VersionTypes) = PackageSpec(;name=n,
 
 has_name(pkg::PackageSpec) = pkg.name !== nothing
 has_uuid(pkg::PackageSpec) = pkg.uuid !== nothing
+has_repo(pkg::PackageSpec) = pkg.repo !== nothing && pkg.repo.url !== nothing
 
 function Base.show(io::IO, pkg::PackageSpec)
     vstr = repr(pkg.version)
-    f = ["name" => pkg.name, "uuid" => has_uuid(pkg) ? pkg.uuid : "", "v" => (vstr == "VersionSpec(\"*\")" ? "" : vstr)]
-    if pkg.repo !== nothing
-        if pkg.repo.url !== nothing
-            push!(f, "url/path" => string("\"", pkg.repo.url, "\""))
-        end
+    f = ["name" => has_name(pkg) ? pkg.name  : "", "uuid" => has_uuid(pkg) ? pkg.uuid : "", "v" => (vstr == "VersionSpec(\"*\")" ? "" : vstr)]
+    if has_repo(pkg)
```

Comment:
Why did you change the logic here? Does not seem to be necessary.

Reply:
I think @fredrikekre is refering to `has_repo`.

`"name" => has_name(pkg) ? pkg.name  : ""` is sufficient to fix the bug.

Reply:
Yup, for example when running `add Example#master`